File: 프로그래머스/unrated/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        int[] answer = new int[queries.length];
        
        for(int i = 0; i < queries.length; i++) {
            int s = queries[i][0];
            int e = queries[i][1];
            int k = queries[i][2];
            
            int ans = 1_000_001;
            for(int j = s; j <= e; j++) {
                if(arr[j] > k) ans = Math.min(ans, arr[j]);
            }
            answer[i] = (ans == 1_000_001) ? -1 : ans;
        }
        
        return answer;
    }
}

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.js
function solution(arr, queries) {
    let answer = [];
    
    queries.forEach((query) => {
        let s = query[0];
        let e = query[1];
        let k = query[2];

        let ans = -1;
        for(let i = s; i <= e; i++) {
            if(arr[i] > k && (arr[i] < ans || ans == -1)) ans = arr[i];
        }
        answer.push(ans);
    })
    return answer;
}